assembly/utils/BytesBuffer.ts

```diff
@@ -34,8 +37,10 @@ export class BytesWriter {
         }
     }
 
-    encodeBool(b: bool): void {
-        this.bytes.push(b ? 0x01 : 0x00);
+    /**
+    * @description Push input in this.bytes
+    */
+    write (input: u8[]): void {
+        Bytes.copy(input, this.bytes, this.bytes.length);
```

Comment:
Why don't we simply use `Array#concat` instead of copy?

Reply:
By concatenation
```
this.bytes = this.bytes.concat(input);

/* Steps
1. const newArray = this.bytes // (By value) -> Copy all elements from bytes to newArray
2. for(i<input.length) this.bytes.push(input[i])
3. Return newArray
*/
```

By our copy
```
Bytes.copy(input, this.bytes, this.bytes.length);
```
/* Steps
1. for(i<input.length) bytes.push(input[i]) 
*/
```